src/Lykke.Logs/Loggers/LykkeSlack/LykkeSlackLoggerProvider.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Collections.Concurrent;
+using JetBrains.Annotations;
+using Microsoft.Extensions.Logging;
+
+namespace Lykke.Logs.Loggers.LykkeSlack
+{
+    [ProviderAlias("Slack")]
+    internal sealed class LykkeSlackLoggerProvider : ILoggerProvider
+    {
+        [NotNull] private readonly ISpamGuard<Microsoft.Extensions.Logging.LogLevel> _spamGuard;
+        [NotNull] private readonly Func<Microsoft.Extensions.Logging.LogLevel, string> _channelResolver;
+
+        [NotNull] private readonly ConcurrentDictionary<string, ILogger> _loggers;
+        [NotNull] private readonly ISlackLogEntriesSender _sender;
+
+        public LykkeSlackLoggerProvider(
+            [NotNull] string azureQueueConnectionString,
```

Comment:
Validate all arguments for null here not in internal classes. Actually  azureQueueConnectionString has no validation

Reply:
Fixed. But usually I prefer to check only variable that class use itself. All that passed to the another classes should be checked by that classes. Only they should know how to check given variable.